_targets.R

```diff
@@ -35,6 +39,103 @@ dir.create("3_visualize/out/daily_timeseries_png/",showWarnings = FALSE)
 dir.create("3_visualize/out/hourly_timeseries_png/",showWarnings = FALSE)
 dir.create("3_visualize/out/nhdv2_attr_png/",showWarnings = FALSE)
 dir.create("3_visualize/out/nhdv2_attr_png/refined",showWarnings = FALSE)
+dir.create("4_predict/out/",showWarnings = FALSE)
```

Comment:
Yeah, that is a lot of new directories 😄. When I create new pipelines, I've actually started taking a different approach where I commit blank text files called `placeholder.txt` so that a user doesn't need to create new directories when they clone a repo, and the `_targets.R` file doesn't get so cluttered. This works well for select, "primary" directories like "2_process/out", but it's probably not practical for creating all of these train/test directories.

At minimum, I think we could cut down on a few of these lines by using `recursive = TRUE` as you mention in your overview comments. To collapse this further and define many new directories at once, my first approach would be to use `expand.grid()` to come up with the different combinations. I tried this briefly and it seems to work (although I have not carefully checked that I've captured all of the directories you have below). The step to define `file_path` using the `mutate` step feels a bit hacky and isn't very scalable, but I ran out of time to play with it for now.

```r
# create directories for each of the following combinations of data splits,
# feature sets and temporal resolutions:
predict_dir <- "4_predict/out"
train_test_other <- c("vip","hypopt") # wasn't sure what hypopt refers to, so just call this vector "other" for now
train_test_splits <- c("temporal","random")
train_test_features <- c("RF_static_dynamic", "RF_min_static_dynamic","RF_dynamic")
train_test_res <- c("pred_obs","spatial_res","monthly_res","annual_res","temporal_res")
rf_xai_plot_types <- c("shap","dependence") 
rf_xai_dep_options <- c("pdp","ice")
train_test_dir <- bind_rows(
  expand.grid(predict_dir, train_test_splits, train_test_res, train_test_features),
  expand.grid(predict_dir, train_test_splits, train_test_other),
  expand.grid(predict_dir, train_test_splits, rf_xai_plot_types[rf_xai_plot_types == "shap"], train_test_features),
  expand.grid(predict_dir, train_test_splits, rf_xai_plot_types[rf_xai_plot_types == "dependence"], train_test_features, rf_xai_dep_options)) %>%
  mutate(file_path = case_when(is.na(Var4) ~ file.path(Var1, Var2, Var3),
                               is.na(Var5) ~ file.path(Var1, Var2, Var3, Var4),
                               TRUE ~ file.path(Var1, Var2, Var3, Var4, Var5))) %>%
  pull(file_path)

# Now actually create the newly-defined directories. 
# You could also use sapply(train_test_dir, dir.create, recursive = TRUE) 
# but I just used a for loop since all we need is to evaluate the expression, 
# rather than return any output.
for(i in seq_along(train_test_dir)){
  dir.create(train_test_dir[i], recursive = TRUE)
}
```

This is more compact, but is also less readable than manually defining each of the new directories on a new line. What do you think?

Reply:
Let me see if what you wrote here generates all the directories. And I can see if the mutate step can be simplified

Reply:
I added 2 items to `train_test_features` that are needed for the `random` split. That generates unnecessary directories for the `temporal` split, but I'm okay with that to have a simpler directory generation method.